BiconeEvolution/current_antenna_evo_build/XF_Loop/Evolutionary_Loop/Loop_Parts/Part_B/Part_B2_Parallel_Pueo.sh

```diff
@@ -0,0 +1,165 @@
+	NC='\033[0m'
+
+	echo -ne "$1"
+	if [ ${#1} -lt 10 ]
+	then
+		for ((i=0; i<10-${#1}; i++))
+		do
+			echo -ne " "
+		done
+	fi
+	echo -ne "[${GREEN}"
+	for ((i=0; i<$num_bars; i++))
+	do
+		echo -ne "#"
+	done
+	echo -ne "${NC}"
+	for ((i=0; i<$num_spaces; i++))
+	do
+		echo -ne "#"
+	done
+	echo -ne "] - flags: $flags, $percent %"
+	echo -ne "\n"
+}
+
+
+cd $WorkingDir
+mkdir -m775 ${XFProj}/XF_models_${gen} 2> /dev/null
+mkdir -m775 Run_Outputs/${RunName}/Root_Files/${gen}_Root_Files 2> /dev/null
+mkdir -m775 Run_Outputs/${RunName}/uan_files/${gen}_uan_files 2> /dev/null
+mkdir -m775 ${PSIMDIR}/outputs/${RunName}/${gen}_outputs 2> /dev/null
+
+# We need to count how many XF and pueoSim jobs are finished
+# We'll do this by counting the number of files in the GPUFlags and PUEOFlags directory
+
+cd $WorkingDir/Run_Outputs/$RunName/GPUFlags
+gpu_flags=$(find . -type f | wc -l)
+cd $WorkingDir/Run_Outputs/$RunName/PUEOFlags
+pueo_flags=$(find . -type f | wc -l)
+cd $WorkingDir/Run_Outputs/$RunName/ROOTFlags
+root_flags=$(find . -type f | wc -l)
+
+
+peuosimsperindiv=49
+peuocount=$((peuosimsperindiv * NPOP))
+max_jobs=250
+job_cutoff=$((max_jobs - peuosimsperindiv))
+UJSER=$(whoami)
+already_checked=0
+
+echo "Waiting for XF and pueoSim jobs..."
+echo ""
+echo ""
+echo ""
+echo ""
+
+# We need to wait until all the XF jobs are finished
+while [[ $root_flags -lt $NPOP ]]
+do
+	# if there is an output file in the TMPGPUFlags directory, 
+	# submit a pueoSim job for that antenna and then move the file to the GPUFlags directory
+	cd $WorkingDir/Run_Outputs/$RunName/TMPGPUFlags
+	# if there are no files in the TMPGPUFlags directory, 
+	# wait 10 seconds and then check again Or if the number of jobs submitted is greater than 250 - 49
+	jobs_submitted=$(squeue -u $USER | wc -l)
+	if [[ $(ls | wc -l) -eq 0 || $jobs_submitted -gt $job_cutoff ]]
+	then
+		sleep 10
+	else
+		for file in *
+		do	
+			# get the antenna number from the file name
+			indiv=$(echo "$file" | cut -d'_' -f5)
+			indiv=$(echo "$indiv" | cut -d'.' -f1)
+			indiv=$(($indiv - $gen*$NPOP))
+			cd $WorkingDir
+			sbatch --array=1-49 --export=ALL,gen=$gen,WorkingDir=$WorkingDir,RunName=$RunName,Seeds=$Seeds,PSIMDIR=$PSIMDIR,NPOP=$NPOP,NNT=$NNT,Exp=$exp,indiv=$indiv --job-name=${RunName} Batch_Jobs/PueoCall_Array_Indiv.sh
+			# move the cursor up 1 line
+			tput cuu 1
+			# move the file to the GPUFlags directory
+			cd $WorkingDir/Run_Outputs/$RunName/TMPGPUFlags
+			mv $file $WorkingDir/Run_Outputs/$RunName/GPUFlags
+		done
+	fi
+	cd $WorkingDir/Run_Outputs/$RunName/GPUFlags
+	gpu_flags=$(ls | wc -l)
+
+	cd $WorkingDir/Run_Outputs/$RunName/PUEOFlags
+	pueo_flags=$(find . -type f | wc -l)
+
+	cd $WorkingDir/Run_Outputs/$RunName/ROOTFlags
+	root_flags=$(find . -type f | wc -l)
+
+	if [ $gpu_flags -eq $XFCOUNT ]
+	then
+	    if [ $already_checked -eq 0 ]
+	    then
+			xf_finish_time=`date +%s`
+			pueosim_start_time=`date +%s`
+			already_checked=1
+		fi
+	fi
+
+	tput cuu 3
```

Comment:
Just out of curiosity, what does this do? Looks like maybe it puts me back to the top of my terminal when I run it on my own terminal.

Reply:
It's so the progress bar I made doesn't move when printing it

Reply:
Gotcha, everything looks good to me then!